compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3971,10 +3972,10 @@ private BType getIntersection(IntersectionContext intersectionContext, BType lhs
                 return intersectionType;
             }
         } else if (type.tag == TypeTags.MAP && lhsType.tag == TypeTags.MAP) {
-            BType intersectionType = createRecordAndMapIntersection(intersectionContext,
-                    type, lhsType, env);
-            if (intersectionType != symTable.semanticError) {
-                return intersectionType;
+            BType intersectionConstraintTypeType = getIntersection(intersectionContext, ((BMapType) lhsType).constraint,
+                                                                   env, ((BMapType) type).constraint);
+            if (intersectionConstraintTypeType != symTable.semanticError) {
+                return new BMapType(TypeTags.MAP, intersectionConstraintTypeType, null);
```

Comment:
Can there be situations where `intersectionConstraintTypeType` can become `null`?

Reply:
Yeah, missed this. Fixed with 3ae949b, thank you!